appeals/api/src/server/endpoints/integrations/integrations.utils.js

```diff
@@ -0,0 +1,38 @@
+/** @typedef {import('pins-data-model').Schemas.AppellantSubmissionCommand['documents'][number]} AppellantSubmissionDocument */
+/** @typedef {import('pins-data-model').Schemas.LPAQuestionnaireCommand['documents'][number]} LPAQuestionnaireCommandDocument */
+
+/**
+ *
+ * @param {(AppellantSubmissionDocument|LPAQuestionnaireCommandDocument)[]} documents
+ * @returns {(AppellantSubmissionDocument|LPAQuestionnaireCommandDocument)[]}
+ */
+export const renameDuplicateDocuments = (documents) => {
+	const seen = new Set();
+
+	return documents.map((document) => {
+		let key = `${document.documentType}_${document.originalFilename}`;
+
+		if (!seen.has(key)) {
+			seen.add(key);
+			return document;
+		}
+
+		const nameParts = document.originalFilename.split('.');
+		const extension = nameParts.pop() || '';
+		const baseName = nameParts.join('.');
```

Comment:
Can we use `node:path` package instead for this logic? I think this logic is fine, but its clearer with methods like `path.basename` and `path.extname`.